src/main/java/core/basesyntax/service/DataConverterServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterServiceImpl implements DataConverterService {
+    private static final String COMMA_SEPARATOR = ",";
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+
+    @Override
+    public List<FruitTransaction> convert(List<String> data) {
+        return data.stream()
+                .filter(FruitTransaction.Operation::startsWithOperation)
+                .map(this::convertToFruitTransaction)
+                .collect(Collectors.toList());
+    }
+
+    private FruitTransaction convertToFruitTransaction(String data) {
+        String[] dataSplit = data.split(COMMA_SEPARATOR);
```

Comment:
Consider adding a check to ensure that `dataSplit` has the expected number of elements before accessing them to prevent `ArrayIndexOutOfBoundsException`.